src/services/findAllReferences.ts

```diff
@@ -2014,16 +2030,40 @@ namespace ts.FindAllReferences {
             readonly symbol: Symbol;
             readonly kind: NodeEntryKind | undefined;
         }
+
+        function isStatic(symbol: Symbol): boolean {
+            if (!symbol?.valueDeclaration) { return false; }
+            if (symbol.valueDeclaration?.modifierFlagsCache) { return !!((symbol.valueDeclaration.modifierFlagsCache & ModifierFlags.Static)); }
+            else { return !!(symbol.valueDeclaration.modifiers?.some((modifier) => modifier.kind & ModifierFlags.Static)); };
+        }
+
         function getRelatedSymbol(search: Search, referenceSymbol: Symbol, referenceLocation: Node, state: State): RelatedSymbol | undefined {
             const { checker } = state;
+            const isReferenceSymbolStatic = isStatic(referenceSymbol);
+            const cbSymbol = (sym: Symbol, rootSymbol: Symbol, baseSymbol: Symbol | undefined, kind: NodeEntryKind): RelatedSymbol | undefined => {
+                // check whether the symbol used to search itself is just the searched one.
+                if (baseSymbol) {
+                    // static method/property and instance method/property might have the same name. Only check static or only check instance.
+                    if ((isReferenceSymbolStatic && !isStatic(baseSymbol)) || (!isReferenceSymbolStatic && isStatic(baseSymbol))) {
+                        baseSymbol = undefined;
+                    }
+                }
+
+                return search.includes(baseSymbol || rootSymbol || sym)

```

Comment:
`includes` is too new, use one of the helper utilities in `core.ts`

Reply:
Oh, the type of search is Search rather than Array.